http/go/nethttp/compression_handler.go

```diff
@@ -0,0 +1,85 @@
+package nethttplibrary
+
+import (
+	"bytes"
+	"compress/gzip"
+	"io"
+	"io/ioutil"
+	"net/http"
+)
+
+type CompressionHandler struct {
+	options CompressionOptions
+}
+
+type CompressionOptions struct {
+	enableCompression bool
+}
+
+func NewCompressionHandler() *CompressionHandler {
+	options := NewCompressionOptions(true)
+	return NewCompressionHandlerWithOptions(options)
+}
+
+func NewCompressionHandlerWithOptions(option CompressionOptions) *CompressionHandler {
+	return &CompressionHandler{options: option}
+}
+
+func NewCompressionOptions(enableCompression bool) CompressionOptions {
+	return CompressionOptions{enableCompression: enableCompression}
+}
+
+func (c *CompressionHandler) Intercept(pipeline Pipeline, middlewareIndex int, req *http.Request) (*http.Response, error) {
+	if c.options.enableCompression != true {
```

Comment:
missing getting any option that might be attached with the request and apply that option value instead if anything is provided.
https://github.com/microsoft/kiota/blob/3d240712a5923c628da5c0faf5560284516d8758/http/go/nethttp/redirect_handler.go#L87